### PR TITLE
Use configured timesteps in DDPM parallel batch steps

### LINKED ISSUE #9261: "previous_timestep()" in DDPM scheduling not compatible with "trailing" option. DDIM bugged too
### Describe the bug

I'm currently on a quest to remove all noise from my very noise-sensitive diffusion application, so checking all the equations right now and I think I found one error:

in scheduling_ddmp.py, previous_timesteps() currently assumes two cases
 - either you have fully custom timesteps, in which case it's properly handled, the prev timestep is taken in the custom array
 - OR, the formula used is:             prev_t = timestep - self.config.num_train_timesteps // num_inference_steps

However, though this formula may be worling for the "leading" scheduler option, it's not correct for "trailing", nor likely for "linspace" since there's a trick with round errors. For example if I use trailing with 1000 training steps and 13 generation steps we have the steps define in set_timesteps: 

tensor([999, 922, 845, 768, 691, 614, 537, 461, 384, 307, 230, 153,  76])

but when the successive steps are actually applied we have:

t, prev_t :
tensor(999) tensor(923) 
tensor(922) tensor(846) 
tensor(845) tensor(769) 
tensor(768) tensor(692) 
tensor(691) tensor(615) 
tensor(614) tensor(538) 
tensor(537) tensor(461) 
tensor(461) tensor(385) 
tensor(384) tensor(308) 
tensor(307) tensor(231) 
tensor(230) tensor(154) 
tensor(153) tensor(77)  
tensor(76) tensor(0)   

Basically, most 'prev_t' are wrong because of the rounding error.

@yiyixuxu and @bghira (from what I've seen about your contribution on 0snr & vprediction you might be relevant for the discussion.)

Also the same formula is used in DDIM so trailing is wrong too. Note that DDIM does not (currently, or maybe ever?) support custom_timesteps.

As a first fix, I believe setting using custom_timesteps=True for "trailing" option should work, but I suspect you may want to have more work to fix this in all schedulers.

Please not that this is quite important since trailing is being recognized as the most relevant way to step, leading and linspace being likely suboptimal for reasons explained in the iddpm and the "Common Diffusion Noise Schedules and Sample Steps are Flawed" paper.

### Reproduction

Literally any model with trailing and a prime number for n_steps.

### Logs

_No response_

### System Info

Ubuntu 22

### Who can help?

@bghira @yiyixuxu

### PR DESCRIPTION
**Pull request by Robertboy18**
# What does this PR do?

I was comparing the DDPM parallel batch path with the regular scheduler step and noticed that `batch_step_no_noise()` still used a fixed training-step stride. This is the remaining batched case from #9261 (Fixes #9261); the scalar path was already fixed in #9384.

For `linspace`, `trailing`, and custom schedules, that could choose different alpha coefficients from the actual next configured timestep. This now follows `scheduler.timesteps` and adds parity tests against the scalar deterministic step :)

I also have a small TorchLean formulation of the schedule-successor invariant and can attach it if useful.

## Tests

- `pytest tests/schedulers/test_scheduler_ddpm_parallel.py -q`: 40 passed
- `make modified_only_fixup`
- `make fix-copies`

## Self-review

- Blocking issues: none.
- Non-blocking issues: none.
- Dead code: none.
- Verdict: READY.

## Before submitting

- [x] I used an AI-assisted workflow.
  - [x] I read the Coding with AI agents guide.
  - [x] I ran the repository self-review against the current diff.
  - [x] The final self-review notes are included above.
- [x] I read the contributor guideline and philosophy doc.
- [x] The existing maintainer-acknowledged discussion is linked above.
- [x] No documentation update is needed because this restores parity with the configured scheduler semantics.
- [x] I added regression tests.

### CODE REVIEW

File: src/diffusers/schedulers/scheduling_ddpm_parallel.py [hunk 1]
```diff
@@ -605,9 +605,14 @@ def batch_step_no_noise(
...
         """
-        t = timesteps
-        num_inference_steps = self.num_inference_steps if self.num_inference_steps else self.config.num_train_timesteps
-        prev_t = t - self.config.num_train_timesteps // num_inference_steps
+        t = timesteps.flatten()
+        if self.custom_timesteps or self.num_inference_steps:
+            schedule = self.timesteps.to(t.device)
+            previous_schedule = torch.cat([schedule[1:], schedule.new_tensor([-1])])
+            schedule_indices = (t[:, None] == schedule[None, :]).to(torch.int64).argmax(dim=-1)
+            prev_t = previous_schedule[schedule_indices]
```
[the remark this answers is not in the document]
Comment (Robertboy18) [reply src/diffusers/schedulers/scheduling_ddpm_parallel.py:616]: Good catch, thanks! I added an explicit no-match check and a regression case for an unexpected timestep :)

File: tests/schedulers/test_scheduler_ddpm_parallel.py [hunk 2]
```diff
@@ -111,6 +113,41 @@ def test_batch_step_no_noise(self):
...
+            timesteps = scheduler.timesteps
+            samples = torch.linspace(-1, 1, timesteps.numel() * 12).reshape(timesteps.numel(), 3, 2, 2)
+            model_outputs = torch.linspace(1, -1, timesteps.numel() * 12).reshape(timesteps.numel(), 3, 2, 2)
+
+            batch_output = scheduler.batch_step_no_noise(model_outputs, timesteps, samples)
+            zero_noise = torch.zeros_like(samples[:1])
+            with mock.patch(
+                "diffusers.schedulers.scheduling_ddpm_parallel.randn_tensor",
+                return_value=zero_noise,
+            ):
```
[the remark this answers is not in the document]
Comment (Robertboy18) [reply tests/schedulers/test_scheduler_ddpm_parallel.py]: Updated the mock to create zeros from the requested shape/device/dtype. The full scheduler test file is now 40/40 locally :)
